Automattic/woocommerce-payments
# Add support for float type values for quantity

Fixes #2904

#### Changes proposed in this Pull Request

Some merchants use plugins such as [Smart Product Quantity](https://woocommerce.com/products/smart-product-quantity/) to sell partial quantities of products. These are handled as floats instead of integers on the product inventory side. When float value is passed through the payment gateway, an error is returned, because the gateway expects integer for quantity.

With my investigation, I have identified the limitation is coming from Stripe's level3 data as it accepts only integers for the `quantity` parameter. Stripe folks shared a potential solution to set the `quantity` to `1` when the type is not an integer. I have tested and I can confirm the order and the payment are successful in that case.

#### Testing instructions

- Ensure Smart [Product Quantity](https://woocommerce.com/products/smart-product-quantity/) is installed and active.
- Enable "Enable global settings for the smart quantity for products." from WP-Admin -> Smart Quantity -> Global Rule.
- Set "Quantity Step" to `0,1` and "Field Style" to "Dropdown".
- Go to the Store page and add a product with quantity `1,5`.
- Try to purchase the product and the following error would appear:
<img width="799" alt="float-quantity-error" src="https://user-images.githubusercontent.com/8667118/136539269-d71c11b8-4d60-4b4e-8f3c-d417979f4b4b.png">

-------------------
